scripts/compose.js

```diff
@@ -1,31 +1,56 @@
+'use strict'
 const fs = require('fs')
-
-const args = process.argv.slice(2)
-const title = args[0]
-const ext = typeof args[1] !== 'undefined' ? args[1] : 'mdx'
-// Remove special characters and replace space with -
-const fileName = title
-  .toLowerCase()
-  .replace(/[^a-zA-Z0-9 ]/g, '')
-  .replace(/ /g, '-')
-  .replace(/-+/g, '-')
-let d = new Date()
-const date = [
-  d.getFullYear(),
-  ('0' + (d.getMonth() + 1)).slice(-2),
-  ('0' + d.getDate()).slice(-2),
-].join('-')
-
-const frontMatter = `---
-title: ${title}
-date: '${date}'
-tags: []
-draft: true
-summary: 
-images: []
----
-`
-
-fs.writeFile(`data/blog/${fileName}.${ext}`, frontMatter, (err) => {
-  if (err) throw err
-})
+var inquirer = require('inquirer')
+inquirer
+  .prompt([
+    {
+      name: 'title',
+      message: 'What shall be the Post Title ?',
+      type: 'input',
+    },
+    {
+      name: 'extention',
+      message: 'What shall be the Post Extention ?',
+      type: 'list',
+      choices: ['md', 'mdx'],
+    },
+    {
+      name: 'draft',
+      message: 'Shall it be a draft ? ( yes/no )',
+      type: 'input',
+    },
+    {
+      name: 'tags',
+      message: 'Any Tags ? Use , to seperate. Leave if no tags',
+      type: 'input',
+    },
+  ])
+  .then((answers) => {
+    // Remove special characters and replace space with -
+    const fileName = answers.title
+      .toLowerCase()
+      .replace(/[^a-zA-Z0-9 ]/g, '')
+      .replace(/ /g, '-')
+      .replace(/-+/g, '-');
+    let d = new Date();
+    const date = [
+      d.getFullYear(),
+      ('0' + (d.getMonth() + 1)).slice(-2),
+      ('0' + d.getDate()).slice(-2),
+    ].join('-');
+    var tagArray = answers.tags.split(',');
+    tagArray.forEach((tag,index) => tagArray[index] = tag.trim());
+    const tags = "'" + tagArray.join("','") + "'"
+    // No Weird Stuff
+    const frontMatter = `---\ntitle: ${answers.title?answers.title:'Untitled'}\ndate: '${date}'\ntags: [${answers.tags?tags:''}]\ndraft: ${answers.draft.toLowerCase() == 'no' ? 'false' : 'true'}\nsummary: \nimages: []\n---`
```

Comment:
This looks really messy. Has this been through prettier? Otherwise, let's abstract it to a function that generates the frontMatter